contracts/dapps/gauge/DappGauge.sol

```diff
@@ -0,0 +1,426 @@
+
+        address factory = IPool(pool).getFactory();
+        
+        gauges[pool].cTokenRevenue = gauges[pool].cTokenRevenue.add(amount);
+
+        emit UpdateLedger(community, factory, pool, amount);
+    }
+
+    function addNewGauge(address community, address pool) external {
+        require(Ownable(community).owner() == msg.sender, "Only community owner can call");
+        require(CommunityFactory(communityFactory).createdCommunity(community), "Invalid community");
+        require(ICommunity(community).poolActived(pool), "Pool is not exist or closed");
+        require(!gauges[pool].hasCreated, "Gauge has added");
+
+        address cToken = ICommunity(community).getCommunityToken();
+        address factory = IPool(pool).getFactory();
+
+        gauges[pool].hasCreated = true;
+        gauges[pool].community = community;
+        gauges[pool].factory = factory;
+        gauges[pool].cToken = cToken;
+
+        emit CreateNewGauge(community, factory, pool);
+    }
+
+    function deposit(address pool, uint256 amount) external nonReentrant {
+        require(gauges[pool].hasCreated, "Gauge not created");
+        if (amount == 0) return;
+        if (!gauges[pool].users[msg.sender].hasDeposited) {
+            gauges[pool].users[msg.sender].hasDeposited = true;
+        }
+
+        _updateNutAcc();
+        _updatePoolAcc(pool);
+
+        address community = gauges[pool].community;
+        address factory = gauges[pool].factory;
+
+        if(gauges[pool].users[msg.sender].amount > 0) {
+            // update user's reward include nut and ctoken
+            uint256 pendingNut = gauges[pool].users[msg.sender].amount.mul(userNutAcc).div(1e12).sub(gauges[pool].users[msg.sender].nutDebt);
+            uint256 pendingCToken = gauges[pool].users[msg.sender].amount.mul(gauges[pool].cTokenAcc).div(1e12).sub(gauges[pool].users[msg.sender].cTokenDebt);
+            gauges[pool].users[msg.sender].nutAvailable = gauges[pool].users[msg.sender].nutAvailable.add(pendingNut);
+            gauges[pool].users[msg.sender].cTokenAvailable = gauges[pool].users[msg.sender].cTokenAvailable.add(pendingCToken);
+        }
+        if (communityTotalStakedVeNut[community] > 0) {
+            // update community's reward only nut
+            uint256 commmunityPending = communityTotalStakedVeNut[community].mul(communityNutAcc).div(1e12).sub(communityDebt[community]);
+            communityAvailable[community] = communityAvailable[community].add(commmunityPending);
+        }
+        if (poolFactoryTotalStakedVeNut[factory] > 0) {
+            // update tool dev's reward only nut
+            uint256 poolFactoryPending = poolFactoryTotalStakedVeNut[factory].mul(poolFactoryNutAcc).div(1e12).sub(poolFactoryDebt[factory]);
+            poolFactoryAvailable[factory] = poolFactoryAvailable[factory].add(poolFactoryPending);
+        }
+
+        // discussion: using transfer or just amount record. veNUT is not transferable
```

Comment:
Maybe we can have a `lock` function in vNUT, locked vNUT can not be used vote again.